File: backend/game-service/src/main/java/com/popplar/gameservice/service/CryptService.java
package com.popplar.gameservice.service;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional(readOnly = true)
public class CryptService {

    @Value("${SALT_A}")
    private Long saltA;
    @Value("${SALT_B}")
    private Long saltB;
    @Value("${SALT_C}")
    private Long saltC;

    public Long encrypt(Long memberId) {
        return memberId * saltA * saltB * saltC;
    }

    public Long decrypt(Long encryptedId) {
        return encryptedId / (saltA * saltB * saltC);
    }
}

File: backend/member-service/src/main/kotlin/com/hotspot/member/service/CryptService.kt
package com.hotspot.member.service

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class CryptService (

    @Value("\${SALT_A}")
    private val saltA: Long,
    @Value("\${SALT_B}")
    private val saltB: Long,
    @Value("\${SALT_C}")
    private val saltC: Long,
){

    fun encrypt(id: Long): Long {
        return id * saltA * saltB * saltC
    }

    fun decrypt(id: Long): Long {
        return id / (saltA * saltB * saltC)
    }
}